### PR TITLE
GRW-1031 / Feature / Format numbers

### PR DESCRIPTION
**Pull request by simonauner**
## What?

Just a minor change to adjust formatting of (larger) numbers now that we have prices that might exceed 1,000.

## Why?

Different languages have different thousand separators. For `sv` the number `1000` is rendered as `"1 000"` but for `en` it's `"1,000"`. This becomes important for pricing as we might render prices above 1,000 for car insurance.

**Ticket(s): [GRW-1031](https://hedvig.atlassian.net/browse/GRW-1031)** (not really specified in the ticket, just a UX consideration from my part)

Swedish:

image

English:

image

### CODE REVIEW

**Review by robinandeer — APPROVED** [review]
Nice, perhaps we can take it one step further and create `useFormatNumber` that automatically gets the locale information. But it's fine to merge like this.

File: src/client/pages/Offer/utils.ts [hunk 1]
```diff
@@ -45,8 +44,8 @@ export const formatRegistrationNumber = (registrationNumber: string) => {
   return registrationNumber.replace(letters, `${letters} `)
 }
 
-export const formatNumber = (number: number, locale: LocaleLabel) => {
+export const formatNumber = (number: number, lang: string) => {
```
Comment (guilhermespopolin) [inline src/client/pages/Offer/utils.ts:47]: nit: maybe call it `localizeNumber`?

**Review by guilhermespopolin — APPROVED** [review]
Noice

File: src/client/pages/Offer/utils.ts (hunk 1 above)
Comment (simonauner) [reply src/client/pages/Offer/utils.ts:47]: Ah, yes, good idea!

File: src/client/l10n/useLocalizeNumber.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  const localizeNumber = useCallback(
+    (number: number) => {
+      if (typeof number !== 'number') return number
+
+      return number.toLocaleString(lang)
+    },
+    [lang],
+  )
+
+  return localizeNumber
```
Comment (robinandeer) [inline src/client/l10n/useLocalizeNumber.ts:32]:
```suggestion
  return useCallback(
    (number: number) => localizeNumber(number, lang),
    [lang],
  )
```

**Review by robinandeer — APPROVED** [review]
Looks good to me, just added a "nit" suggestion to keep things DRY ;)

File: src/client/l10n/useLocalizeNumber.ts (hunk 2 above)
Comment (simonauner) [reply src/client/l10n/useLocalizeNumber.ts]: Ah, good thinking! Fixed it and added a test as well

**Review by robinandeer — APPROVED** [review]

**Pull request merged**